.github/workflows/notebook-testing.yaml

```diff
@@ -62,5 +62,14 @@ jobs:
           notebooks=$(cat .github/notebook_lists/vanilla_notebooks.txt | grep -v '^#' | tr '\n' ' ')  
           for notebook in $notebooks; do 
             echo "Executing $notebook"  
-            jupyter nbconvert --to notebook --execute --inplace $notebook  
+            rm -rf .venv
+            python -m venv .venv
+            ls -l `which jupyter`
+            source .venv/bin/activate
+            ls -l `which jupyter`
```

Comment:
Before and after activation we get the same jupyter:
```
-rwxrwxrwx+ 1 runner docker 253 Oct 27 20:17 /opt/hostedtoolcache/Python/3.10.15/x64/bin/jupyter
-rwxrwxrwx+ 1 runner docker 253 Oct 27 20:17 /opt/hostedtoolcache/Python/3.10.15/x64/bin/jupyter
```

But notably it isn't the jupyter the test is failing to find:
```[Errno 13] Permission denied: '/usr/local/share/jupyter'```

Reply:
You need to create the venv in one of the first steps of the job before the pip install, and in each subsequent step using python, you need to source the activate command first since each step is a fresh bash. That is, activating the venv in one step does not mean it is active in any other step.

Reply:
Thanks for the suggestion @bjhargrave. It turns out `ipykernel install` was trying to write to an inaccessible location (`/usr/local/share`). I fixed it by adding `--user` to the command to get it to write to a location accessibly by the current user.

In this change, I'm looking to isolate each notebook execution to its own virtual env.